src/systems/filecoin_vm/interpreter/vm_interpreter.go

```diff
@@ -285,72 +280,58 @@ func _gasToFIL(gas msg.GasAmount, price msg.GasPrice) actor.TokenAmount {
 	// return actor.TokenAmount(util.UVarint(gas) * util.UVarint(price))
 }
 
-// Builds a message for paying block reward from the treasury account to a miner owner.
-func _makeBlockRewardMessage(
-	state st.StateTree, minerOwnerAddr addr.Address, minerPenaltyTotal actor.TokenAmount) msg.UnsignedMessage {
-
-	var blockReward actor.TokenAmount
-	TODO() // TODO: finish
-
-	blockReward -= minerPenaltyTotal
-	if blockReward < 0 {
-		blockReward = 0
+// Builds a message for paying block reward to a miner's owner.
+func _makeBlockRewardMessage(state st.StateTree, minerAddr addr.Address, penalty actor.TokenAmount) msg.UnsignedMessage {
+	params := make([]util.Serialization, 2)
+	params[0] = addr.Serialize_Address(minerAddr)
+	params[1] = actor.Serialize_TokenAmount(penalty)
+
+	sysActor := state.GetActorState(addr.SystemActorAddr)
+	return &msg.UnsignedMessage_I{
+		From_:       addr.SystemActorAddr,
+		To_:         addr.RewardActorAddr,
+		Method_:     sysactors.Method_RewardActor_AwardBlockReward,
+		Params_:     params,
+		CallSeqNum_: sysActor.CallSeqNum(),
+		Value_:      0,
+		GasPrice_:   0,
+		GasLimit_:   msg.GasAmount_SentinelUnlimited(),
 	}
-
-	// networkTreasuryActor := loadActor(NetworkTreasuryActorAddress)
-	// minerActor := loadActorSubstate(minerActorAddr)
-	// minerWorker := loadActor(minerActor.Info.Worker)
-	//return &UnsignedMessage_I{
-	//	From_:       minerActor.Info.Worker,
-	//	To_:         NetworkTreasuryActorAddress,
-	//	Method_:     NetworkTreasury.PayBlockReward,
-	//	Params_:     serialize([minerActorAddr]),
-	//	CallSeqNum_: minerWorker.CallSeqNum,
-	//	Value_:      0,
-	//	GasPrice_:   0,
-	//	GasLimit_:   payBlockRewardGasLimit,
-	//}
-	panic("TODO: implement when network treasury actor implemented")
 }
 
 // Builds a message for submitting ElectionPost on behalf of a miner actor.
-func _makeElectionPoStMessage(state st.StateTree, minerActorAddr addr.Address, epoch UInt64, postProof Bytes) msg.UnsignedMessage {
-	// minerActor := loadActorSubstate(minerActorAddr)
-	// minerWorker := loadActor(minerActor.Info.Worker)
-	//return &UnsignedMessage_I{
-	//	From_:       minerActor.Info.Worker,
-	//	To_:         minerActorAddr,
-	//	Method_:     StorageMinerActor.SubmitElectionPoSt,
-	//	Params_:     serialize([PoStSubmission{postProof, epoch}]),
-	//	CallSeqNum_: minerWorker.CallSeqNum,
-	//	Value_:      0,
-	//	GasPrice_:   0,
-	//	GasLimit_:   sumbitElectionPostGasLimit,
-	//}
-	panic("TODO: implement when necessary dependencies are importable")
+func _makeElectionPoStMessage(state st.StateTree, minerActorAddr addr.Address, epoch UInt64) msg.UnsignedMessage {
+	// TODO: determine parameters necessary for this message.
```

Comment:
What input is needed (proofs convo end?)?

Reply:
You tell me!

Reply:
will handle in #724 once alignment is found